### PR TITLE
Fix LLVM feature check

### PR DESCRIPTION
**Pull request by wsmoses**

### CODE REVIEW

**Comment by vchuravy** [comment]
Fixes #1580

File: src/compiler/optimize.jl [hunk 1]
```diff
@@ -1937,8 +1937,12 @@ function optimize!(mod::LLVM.Module, tm)
         basic_alias_analysis!(pm)
         cfgsimplification!(pm)
         dce!(pm)
-@static if isdefined(GPUCompiler, :cpu_features!)
+@static if isdefined(LLVM, :cpu_features!)
+        LLVM.cpu_features!(pm)
+else
+@staticif isdefined(GPUCompiler, :cpu_features!)
```
Comment (avik-pal) [inline src/compiler/optimize.jl:1943]:
```suggestion
@static if isdefined(GPUCompiler, :cpu_features!)
```

File: src/compiler/optimize.jl [hunk 2]
```diff
@@ -1937,8 +1937,12 @@ function optimize!(mod::LLVM.Module, tm)
         basic_alias_analysis!(pm)
         cfgsimplification!(pm)
         dce!(pm)
+@static if isdefined(LLVM, :cpu_features!)
+        LLVM.cpu_features!(pm)
+else
 @static if isdefined(GPUCompiler, :cpu_features!)
```
Comment (vchuravy) [inline src/compiler/optimize.jl:1943]:
```suggestion
elseif isdefined(GPUCompiler, :cpu_features!)
```

?

Comment (wsmoses) [reply src/compiler/optimize.jl:1943]: does static still apply to the else tho?

that was my worry

Comment (vchuravy) [reply src/compiler/optimize.jl:1943]: Yeah it should

**Pull request merged**